File: questions/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i in range(0,len(nums)):
            le = target - nums[i]
            if le in dic:
                return [dic[le] , i]
            else:
                dic[nums[i]] = i

File: questions/two_sum.go
func twoSum(nums []int, target int) []int {
    dic := make(map[int] int)
    for idx , num := range nums{
        num2 := target - num 
        if _ ,ok := dic[num2] ; ok && idx != dic[num2]{
            return [] int {dic[num2] ,idx} 
        }
        dic[num] = idx
    }
    return [] int {-1,-1}
}
